### DISCUSSION TITLE
We are planning to archive this repository

### DISCUSSION THREAD

**Discussion by jtcohen6** [dbt-labs/dbt-presto#83] [Announcements]
**We are planning to archive the `dbt-presto` repository. We are not planning to release `dbt-presto==1.0.0`.**

Thanks to those of you who have given us feedback over the past few years, and apologies to those who have opened issues or PRs over the past few months, while we were reaching this decision. The adapter plugin experiment, of which `dbt-presto` was first test subject, has been an overwhelming success. It's now time for `dbt-presto` to take its place in the archives.

### **Why this decision?**

Our team lacks hands-on expertise using dbt with Presto, at scale and in production—and so we also lack the hard-won, mettle-tested opinions that we’ve encoded into the dbt plugins we maintain for other data platforms.

The current `dbt-presto` plugin started as the _very first_ adapter, way back in 2019 (https://github.com/dbt-labs/dbt-core/issues/1106). Its initial purpose was to prove the value of the adapter interface, which now enables community members to use dbt with dozens of different databases, query engines, and SQL-speaking data platforms.

Since its origin, `dbt-presto` has been lacking next-level capabilities—incremental modeling, snapshots, efficient-at-scale metadata access—that other adapters have been able to offer. We don’t believe we’re the right people to bring it up to feature parity with those other plugins.

We *are* the right people to continue making foundational investments in `dbt-core` and the adapter interface. To that end, we’re going to continue supporting the maintainers of new and existing open source adapter plugins.

### **Why now?**

We believe it's 100% possible to upgrade `dbt-presto` for compatibility with `dbt-core==1.0.0`. It might not even require that much work.

The v1.0 release of dbt Core is our opportunity to communicate intent—quality, maturity, and stability over the long term—and we’re signalling that intent for each adapter plugin that we’ve upgraded alongside `dbt-core`. For `dbt-presto`, it’s not a commitment we can make in good faith going forward.

### **What are the alternatives?**

We’ve been working closely with the team at Starburst Data, who are now officially maintaining the open source `dbt-trino` adapter. Hopefully, this adapter offers compatibility (or an easy migration) for many current users of the `dbt-presto` adapter. Check out:

- [Repository: starburstdata/dbt-trino](https://github.com/starburstdata/dbt-trino)
- [Announcement on Starburst blog](https://docs.starburst.io/blog/2022-01-03-dbt-trino.html)
- [Documentation: dbt + Trino](https://docs.getdbt.com/reference/warehouse-profiles/trino-profile)

If you’re someone who wants or needs to keep using `dbt-presto` in its current form, we invite you to fork this repository. Comment in the threads below, and we can share information (such as docs links or sample code) to assist in upgrading the plugin for use with dbt Core v1.0.

If you’re someone with real experience deploying dbt with Presto in production, and an interest in giving back to the dbt community as the maintainer of an open source adapter plugin: let’s talk!

**Comment by leniartek** [comment]
Happy to answer any questions regarding `dbt-trino` and Starburst commitment to support this open source adapter!

**Comment by rikturr** [comment]
If someone were to do the work to make `dbt-presto` compatible with `dbt-core==1.0.0`, would that be a reason to un-archive this repo? Or would it live somewhere else as a community adapter?

**Reply by rikturr** [reply]
Welp, didn't read the whole post. Looks like the last sentence indicates an open-source community adapter

**Reply by leniartek** [reply]
Yes, `dbt-trino` is open source.
https://github.com/starburstdata/dbt-trino

**Reply by jtcohen6** [reply]
@rikturr That's our current thinking. We'd be excited about this repo living elsewhere, as a community-maintained adapter. I'll let you know (and update the discussion post) if any part of that changes. In the meantime, please let me know if you need any help with the upgrade for v1 compatibility—I'd be happy to put together a sample PR, for instance.
